src/pages/Dashboard/Modals/SupplyWithdraw/SupplyWithdrawForm.tsx

```diff
@@ -77,23 +88,41 @@ export const SupplyWithdrawContent: React.FC<
       const temp = calculateNewBalance(amountInUsd);
       updateBorrowLimit = BigNumber.maximum(temp, 0);
     }
-
-    return [updateBorrowLimit];
+    const updateBorrowLimitCents = updateBorrowLimit?.multipliedBy(100);
+    return [updateBorrowLimitCents];
   }, [amount, asset?.id, userTotalBorrowBalance, userTotalBorrowLimit]);
 
+  const dailyEarningsCents = useMemo(() => {
+    const hypotheticalAssets = [...assets];
+    // Modify asset with hypotheticalBalance
+    if (amount) {
+      const hypotheticalAsset = { ...asset, supplyBalance: asset.supplyBalance.plus(amount) };
```

Comment:
This will cause an issue in the withdraw tab, as it will consider the `amount` as being additional supply while we're actually withdrawing from the supply.

I think we unfortunately have the same issue with the calculation of `hypotheticalBorrowLimitCents` above.

Reply:
I refactored `calculateNewBalance` to accept an initial value and am using it now here and for the supply Balance. `hypotheticalBorrowLimitCents` should be good because it is already using this function